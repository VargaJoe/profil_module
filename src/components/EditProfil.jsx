import React, { Component } from 'react';
import Avatar from './Avatar';

class EditProfil extends Component {
	render () {

		let user_info = this.props.user.infoList.map((info, index) =>
			<div key={index} className="user__edit_global_info__item">
				<label htmlFor="">{info.name}</label>
				<input id="" type="text" placeholder={info.name} defaultValue={info.value}/>
			</div>
		);

		return (
			<div className="profil">
				<div className="user" >
					<Avatar />
					<div className="user__global_info">
						<fieldset>
							<legend>User name</legend>
							<input id="userName" type="text" placeholder="User name" defaultValue={this.props.user.name} />
						</fieldset>
						<fieldset>
							<legend>Position</legend>
							<input id="userPosition" type="text" placeholder="Position"  defaultValue={this.props.user.position} />
						</fieldset>
						<fieldset>
							<legend>User Info</legend>
							<div className="user__edit_global_info">
								{user_info}
							</div>
						</fieldset>
					</div>
				</div>

				<fieldset>
					<legend>{this.props.user.aboutTitle}</legend>
					<textarea id="userAbout" rows="10" cols="80" defaultValue={this.props.user.about}>

					</textarea>
				</fieldset>


				<button className="btn" onClick={this.props.action} >
					Save Changes
				</button>
			</div>
		)
	}
};

export default EditProfil;
