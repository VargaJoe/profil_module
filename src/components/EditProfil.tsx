import * as React							from 'react';
import { connect }              			from 'react-redux';
import UserAvatar 							from './UserAvatar';
import { Link } 							from 'react-router-dom';
import { Actions }                			from '@sensenet/redux';
import { PathHelper }                       from '@sensenet/client-utils';
import { User } 							from '@sensenet/default-content-types';
// import { IODataParams } 					from '@sensenet/client-core/dist/Models/IODataParams';

// save config 
const DATA = require('../config.json');

export interface Props {
	updateUser: 		Function;
	saveChanges:  		Function;
	updateUserAvatar:	Function;
	user: 				any;
}

class EditProfil extends React.Component<Props, any> {

	constructor(props: any) {
		super(props);
		this.state = {
			imageIsChanged: ''
		};
	}

	onUpdateImageChanges = (val: any) => {
		this.setState({
			imageIsChanged: val
		});
	}

	render () {
		console.log(this.state.imageIsChanged);
		// *********************************************
		// user info 
		let FullNameInput: 			HTMLInputElement;
		let JobTitleInput: 			HTMLInputElement;
		let EmailInput: 			HTMLInputElement;
		let LanguagesInput: 		HTMLInputElement;
		let PhoneInput: 			HTMLInputElement;
		let BirthDateInput: 		HTMLInputElement;
		let EducationInput: 		HTMLInputElement;
		let DescriptionInput: 		HTMLTextAreaElement;
		// *********************************************

		let userUpdate: any;

		// let image = '';
		// if ( this.state.imageIsChanged.newImage && this.state.imageIsChanged.isChanged ) {
		// 	image = this.state.imageIsChanged.imageSource;
		// }

		const onSaveChanges = (e: any) => {
			let user = {
				Id: 			this.props.user.Id,
				FullName: 		FullNameInput.value,
				JobTitle: 		JobTitleInput.value,
				Email: 			EmailInput.value,
				Languages: 		LanguagesInput.value,
				Phone: 			PhoneInput.value,
				BirthDate: 		BirthDateInput.value,
				Education: 		EducationInput.value,
				Description: 	DescriptionInput.value,
				ImageRef: 	 	5736
			} as User;

			// console.log(this.state.imageIsChanged.name);

			// update user info in sensenet app
			let path = PathHelper.joinPaths(DATA.ims, this.props.user.Name);
			
			userUpdate = this.props.updateUser(path, user);
			
			userUpdate.then( (result: any) => {
				console.log('Update success');
			});
	
			userUpdate.catch((err: any) => {
				console.log('Error success');
			});

			this.props.updateUserAvatar('/Root/Sites/Profil/Avatar', this.state.imageIsChanged, 'Image');

			// this.props.saveChanges(user);
			
		};

		return (
			<div className="profil">

				<UserAvatar 
					onUpdate={this.onUpdateImageChanges}
				/>

				<div className="user" >
					
					<div className="user__global_info">
						<fieldset>
							<legend>User name</legend>
							<input 
								id="userName" 
								type="text" 
								ref={(input) => FullNameInput = input as HTMLInputElement} 
								placeholder="User name" 
								defaultValue={this.props.user.FullName}
								required={true}
							/>
						</fieldset>
						<fieldset>
							<legend>Position</legend>
							<input 
								id="userPosition" 
								type="text" 
								ref={(input) => JobTitleInput = input as HTMLInputElement} 
								placeholder="Position"  
								defaultValue={this.props.user.JobTitle}
								required={true}
							/>
						</fieldset>
						<fieldset>
							<legend>User Info</legend>
							<div className="user__edit_global_info">
								<div className="user__edit_global_info__item">
									<label htmlFor="">Email</label>
									<input 
										id="" 
										type="text" 
										ref={(input) => EmailInput = input as HTMLInputElement} 
										placeholder="Email" 
										defaultValue={this.props.user.Email}
										required={true}
									/>
								</div>
								<div className="user__edit_global_info__item">
									<label htmlFor="">Languages</label>
									<input 
										id="" 
										type="text" 
										ref={(input) => LanguagesInput = input as HTMLInputElement} 
										placeholder="Languages" 
										defaultValue={this.props.user.Languages}
									/>
								</div>
								<div className="user__edit_global_info__item">
									<label htmlFor="">Phone</label>
									<input 
										id="" 
										type="text" 
										ref={(input) => PhoneInput = input as HTMLInputElement} 
										placeholder="Phone" 
										defaultValue={this.props.user.Phone}
									/>
								</div>
								<div className="user__edit_global_info__item">
									<label htmlFor="">BirthDate</label>
									<input 
										id="" 
										type="text" 
										ref={(input) => BirthDateInput = input as HTMLInputElement} 
										placeholder="BirthDate" 
										defaultValue={this.props.user.BirthDate}
									/>
								</div>
								<div className="user__edit_global_info__item">
									<label htmlFor="">Education</label>
									<input 
										id="" 
										type="text" 
										ref={(input) => EducationInput = input as HTMLInputElement} 
										placeholder="Education" 
										defaultValue={this.props.user.Education}
									/>
								</div>
							</div>
						</fieldset>
					</div>
				</div>

				<fieldset>
					<legend>About</legend>
					<textarea id="userAbout" ref={(input) => {DescriptionInput = input as HTMLTextAreaElement; }} defaultValue={this.props.user.Description} />
				</fieldset>
				
				<Link to={'user:' + this.props.user.Name} className="sn_btn" onClick={e => { onSaveChanges(e); }}>
					Save Changes
				</Link>
			</div>
		);
	}
}

const mapStateToProps = (state: any, match: any) => {
	return {
	  user : 			state.user.user,
	  userAvatar :  	state.user.user.Avatar._deferred,
	};
  };

export default connect(
	mapStateToProps,
	(dispatch) => ({
		// saveChanges:  			(userInfo: any) => dispatch({ type: 'SET_USER_INFO', payload: userInfo }),
		updateUser:    			(path: string, options: User) => dispatch(Actions.updateContent( path, options )),
		updateUserAvatar:  		(parentPath: string, file: any, contentType: string) => dispatch(Actions.uploadRequest( parentPath, file, contentType)),
	})
)(EditProfil);
