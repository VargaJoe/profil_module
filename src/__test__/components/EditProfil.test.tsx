import * as React 						  		from 'react';
import EditProfil 							    from '../../components/EditProfil';
import { 
	configure, 
		shallow,
		mount, 
		ShallowWrapper, 
		ReactWrapper } 							from 'enzyme';
import * as Adapter 					  		from 'enzyme-adapter-react-16';
import toJson 				      				from 'enzyme-to-json';
import * as createMockStore						from 'redux-mock-store';
import { Provider } 							from 'react-redux';
import {
    BrowserRouter as Router
} 												from 'react-router-dom';
import { PathHelper } 							from '@sensenet/client-utils';

configure( {adapter: new Adapter()} );

const DATA = require('../../config.json');

describe('<EditProfil /> shallow rendering', () => {
	let store;
	let editprofileShallow: ShallowWrapper<any, any>;
	let editprofileMount: ReactWrapper<any, any>;
	const mockStore = (createMockStore as any)();
	beforeEach( () => {
        store = mockStore({
			sensenet: {
				session: {
					loginState: 'Unauthenticated',
					user: {
						userName: 'Visitor'
					}
				}
            },
			user: {
				user: {
					Id: 12312,
					FullName: 		'adasdasdasd',
					JobTitle: 		'adasdasdasd',
					Email: 			'adasdasdasd',
					Languages: 		'adasdasdasd',
					Phone: 			'adasdasdasd',
					BirthDate: 		'adasdasdasd',
					Education: 		'adasdasdasd',
					Description: 	'adasdasdasd',
				}
			}
		});
		editprofileShallow = shallow(
		<Provider store={store}>
			<EditProfil/>
		</Provider>); 
		editprofileMount = mount(
			<Router>
				<Provider store={store}>
					<EditProfil/>
				</Provider>
			</Router>);
	});
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(editprofileMount)).toMatchSnapshot();
	});
});

// 	it('Test saveChanges  ', () => {
// 		let userWithAvatar = {
// 			// !important user id
// 			Id: 			3211,
// 			AvatarImageRef: 'adasdasda'
// 		};
// 		const fff = editprofileMount
// 		.children()
// 		.children()
// 		.children()
// 		.children();
// 		const saveChanges = fff.props().saveChanges(userWithAvatar);
// 		expect(saveChanges.type).toEqual('SET_USER_INFO');
// 		expect(saveChanges.payload).toEqual(userWithAvatar);
// 	});

// 	it('Test updateUserSN  ', () => {
// 		let userWithAvatar = {
// 			// !important user id
// 			Id: 			3211,
// 			AvatarImageRef: 'adasdasda'
// 		};
// 		const fff = editprofileMount.children()
// 		.children()
// 		.children()
// 		.children();
// 		let pathToUser = PathHelper.joinPaths(DATA.ims, 'Freeman');
// 		const getUsers = fff.props().updateUserSN(pathToUser, userWithAvatar);
// 		expect(getUsers.type).toEqual('UPDATE_CONTENT');
// 	});

// 	it('test onUpdateImageChanges', () => {
		
// 		const fff = editprofileMount
// 		.children()
// 		.children()
// 		.children()
// 		.children();
// 		const spy = jest.spyOn(fff.instance(), 'onUpdateImageChanges');
// 		const ddd = fff.instance().onUpdateImageChanges('lol');
// 		expect(spy).toHaveBeenCalledTimes(1);
// 		expect(spy).toBeCalledWith('lol');
// 	});

// 	it('Test updateUserAvatar  ', () => {
// 		let userWithAvatar = {
// 			// !important user id
// 			Id: 			3211,
// 			AvatarImageRef: 'adasdasda'
// 		};
// 		const fff = editprofileMount
// 		.children()
// 		.children()
// 		.children()
// 		.children();
// 		const getUsers = fff.props().updateUserAvatar(DATA.avatar, 'freeman', 'Image');
// 		expect(getUsers.type).toEqual('UPLOAD_CONTENT');
// 	});

// 	// it('test onSaveChanges', () => {
		
// 	// 	const fff = editprofileMount
// 	// 	.children()
// 	// 	.children()
// 	// 	.children()
// 	// 	.children();
// 	// 	const spy = jest.spyOn(fff.instance(), 'onSaveChanges');
// 	// 	const ddd = fff.instance().onSaveChanges();
// 	// 	expect(spy).toHaveBeenCalledTimes(1);
// 	// });
		
// });

// describe('<EditProfil /> shallow rendering with empty user', () => {
// 	let store;
// 	let editprofileMount: ReactWrapper<any, any>;
// 	const mockStore = (createMockStore as any)();
// 	beforeEach( () => {
//         store = mockStore({
// 			sensenet: {
// 				session: {
// 					loginState: 'Unauthenticated',
// 					user: {
// 						userName: 'Visitor'
// 					}
// 				}
//             },
// 			user: {
// 				user: null
// 			}
// 		});
// 		editprofileMount = mount(
// 			<Router>
// 				<Provider store={store}>
// 					<EditProfil/>
// 				</Provider>
// 			</Router>);
// 	});
// 	// test Snapshot 
// 	it('Match to snapshot', () => {
// 		expect(toJson(editprofileMount)).toMatchSnapshot();
// 	});
		
// });
