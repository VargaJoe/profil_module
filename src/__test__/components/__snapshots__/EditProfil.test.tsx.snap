// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfil /> shallow rendering Match to snapshot 1`] = `undefined`;
